## Statistical Analysis of Starbucks Nutritional Drink Data Updated with Pandas
Julia Beitel - Big Data - October 16, 2018

### Abstract 
In Lab 3, I will be editing lab 2 with newfound data cleaning information on Pandas. In this lab I will be analyizing the Starbucks Nutritional Drink data. Within my analysis, I will be answering the following questions that the data posed: Which drink has the most and least of each nutrional section? How many different drinks are there? What percentage of drinks have 2% milk?

I added (new analysis and cleaning) to lab 3, what are the mean values from each data section? and I attempted to fix the dietary fiber column header. I also replaced all of my code with simpler code using pandas data package. 

### Dataset Preparation
The given data represents the following columns: Beverage category, beverage name, beverage preparation, calories, total fat, trans fat, saturated fat, sodium, total carbs, cholestorol, dietary fiber, sugar, protein, vitamin A %, vitamin C %, iron %, and caffeine.
This data was uploaded on July 30th of 2018 from the following source: https://www.kaggle.com/starbucks/starbucks-menu

#### Import Packages

In [66]:
#Import modules to use the functions and methods they contain
import math as m
import numpy as np
import scipy as sp
import pandas as pd
import matplotlib as mpl
import matplotlib.pyplot as plt
import seaborn as sns

#### Open and Download

In [67]:
#NEW one line to open and read my csv file into a data frame data structure
drink_data = pd.read_csv('starbucks_drinkMenu_expanded.csv')
drink_data.head()

Beverage_category       Beverage      Beverage_prep  Calories  \
0                   Coffee  Brewed Coffee              Short         3   
1                   Coffee  Brewed Coffee               Tall         4   
2                   Coffee  Brewed Coffee             Grande         5   
3                   Coffee  Brewed Coffee              Venti         5   
4  Classic Espresso Drinks    Caffè Latte  Short Nonfat Milk        70   

   Total Fat (g)  Trans Fat (g)   Saturated Fat (g)   Sodium (mg)  \
0            0.1             0.0                0.0             0   
1            0.1             0.0                0.0             0   
2            0.1             0.0                0.0             0   
3            0.1             0.0                0.0             0   
4            0.1             0.1                0.0             5   

    Total Carbohydrates (g)   Cholesterol (mg)   Dietary Fibre (g)  \
0                          5                 0                   0   
1                         10                 0                   0   
2                         10                 0                   0   
3                         10                 0                   0   
4                         75                10                   0   

    Sugars (g)   Protein (g)  Vitamin A (% DV)  Vitamin C (% DV)  \
0            0            0.3                0%               0%   
1            0            0.5                0%               0%   
2            0            1.0                0%               0%   
3            0            1.0                0%               0%   
4            9            6.0               10%               0%   

   Calcium (% DV)  Iron (% DV)  Caffeine (mg)  
0               0%           0%           175  
1               0%           0%           260  
2               0%           0%           330  
3               2%           0%           410  
4              20%           0%            75

In [68]:
#NEW Tried to rename header of dietary fiber
#df.rename(columns={
#    'Dietary Fibre (g)': 'Dietary Fiber (g)'},
#    inplace=True)
#error message says there is no .rename() fxn

### Data Modeling 
First, I found the drink with the highest and lowest numbers of nutritional data in each nutritional section...

#### Min and Max Nutritional Data

In [69]:
#NEW I found the min and max of the drink data simply with the .describe() fxn
value = drink_data.describe()
value

#I do not know how I am supposed to connect each item in the list with it's specific index name
#Without this information, I am unable to name the actual drink names for each min and max of nutriitional data

Calories  Trans Fat (g)   Saturated Fat (g)   Sodium (mg)  \
count  242.000000      242.000000         242.000000    242.000000   
mean   193.871901        1.307025           0.037603      6.363636   
std    102.863303        1.640259           0.071377      8.630257   
min      0.000000        0.000000           0.000000      0.000000   
25%    120.000000        0.100000           0.000000      0.000000   
50%    185.000000        0.500000           0.000000      5.000000   
75%    260.000000        2.000000           0.100000     10.000000   
max    510.000000        9.000000           0.300000     40.000000   

        Total Carbohydrates (g)   Cholesterol (mg)   Dietary Fibre (g)  \
count                 242.000000        242.000000          242.000000   
mean                  128.884298         35.991736            0.805785   
std                    82.303223         20.795186            1.445944   
min                     0.000000          0.000000            0.000000   
25%                    70.000000         21.000000            0.000000   
50%                   125.000000         34.000000            0.000000   
75%                   170.000000         50.750000            1.000000   
max                   340.000000         90.000000            8.000000   

        Sugars (g)   Protein (g)   
count   242.000000     242.000000  
mean     32.962810       6.978512  
std      19.730199       4.871659  
min       0.000000       0.000000  
25%      18.000000       3.000000  
50%      32.000000       6.000000  
75%      43.750000      10.000000  
max      84.000000      20.000000

#### Mean Number of Nutritional Data

In [70]:
#NEW found the mean number of each section of nutritional data by the .describe() fxn
#I found that:
#mean number of calories is 193.871901
#mean number of trans fat is 1.307025 g
#mean number of saturated fat is 0.037603 g
#mean number of sodium is 6.363636 mg
#mean number of carbs is 128.884298 g
#mean number of cholesterol is 35.991736 mg
#mean number of dietary fiber is 0.805785 g
#mean number of sugars is 32.962810 g
#mean number of prtein is 6.978512 g

#### Number of Total Drinks

In [71]:
#NEW Next, I found the total number of drinks with the .count() fxn 
#Table shows that there are 242 total drinks on the menu
drink_data.count()

Beverage_category            242
Beverage                     242
Beverage_prep                242
Calories                     242
 Total Fat (g)               242
Trans Fat (g)                242
Saturated Fat (g)            242
 Sodium (mg)                 242
 Total Carbohydrates (g)     242
Cholesterol (mg)             242
 Dietary Fibre (g)           242
 Sugars (g)                  242
 Protein (g)                 242
Vitamin A (% DV)             242
Vitamin C (% DV)             242
 Calcium (% DV)              242
Iron (% DV)                  242
Caffeine (mg)                241
dtype: int64

#### Percent of 2% Milk Drinks

I wasn't sure how to go about finding the percentage of milk drinks with pandas. Because I did not title my column headers as new names, there is not a way that I know of where I can index or search for each specific drinks placement on the graph. Due to this issue, I do not know how I am supposed to total the amount of drinks with 2% milk in their beverage prep table without using an if-else statement or a for loop. 

### Data Analysis 

With my modeling, I still found that:
- the drink with the most calories had 510
- the drink with the most trans fat had 9 g
- the drink with the most saturated fat had 0.3 g
- the drink with the most sodium had 40 mg
- the drink with the most carbs had 340 g
- the drink with the most cholesterol had 90 mg
- the drink with the most dietary fiber had 8 g
- the drink with the most sugar had 84 g
- the drink with the most protein had 20 g

These numbers are reasonable because there are no apparent outliers in the data.

- the drink with the least calories had 0
- the drink with the least trans fat had 0 g
- the drink with the least saturated fat had 0 g
- the drink with the least sodium had 0 mg
- the drink with the least carbs had 0 g
- the drink with the least cholesterol had 0 mg
- the drink with the least dietary fiber had 0 g
- the drink with the least sugar had 0 g
- the drink with the least protein had 0 g

This makes valid sense because some drinks, like black coffee, are made with no addedd ingredients.

I also found that:

- the mean number of calories is 193.871901
- the mean number of trans fat is 1.307025 g
- the mean number of saturated fat is 0.037603 g
- the mean number of sodium is 6.363636 mg
- the mean number of carbs is 128.884298 g
- the mean number of cholesterol is 35.991736 mg
- the mean number of dietary fiber is 0.805785 g
- the mean number of sugars is 32.962810 g
- the mean number of protein is 6.978512 g

Again, I was unable to figure out how to access the specific names of each drink so I could only find the values of the drinks, rather than their corresponding names. Additionally, I could not find the percentage of drinks that included 2% milk due to my inability to count items in a list based on their index and certain values on a table. 

### Conclusion

For further exploration, I would like to continue to figure out some of the issues I had with this dataset. I would like to learn to call the specific names of drinks when I found their specific nutritional values. I would also like to learn how to count the total number of 2% milk prepped drinks in this way. If I had more time, I would also like to be able to drop the first column entirely so that each drink was listed by it's name (which could be changed) and not a number from the list in the first column. Furthermore, I would have liked to be able to change the header name but for some reason, by using the same code that worked in Lab 4 on the candy data, Lab 3 would not stop giving me the same error message. 

This data also raises many questions. Some of them being: Is it hard to dine at Starbucks and be vegan? What does the food nutritional data show? What is the most unhealthy meal possible at Starbucks? 

The data also posed questions towards the validity and legitamacy of the data collection, as it did in the last lab. Some being: Are there any flaws in the data? How accurate are these numbers? Has the nutritional data changed since this data was collected? 
It would be interesting to dive deeper into these questions because, living in Seattle, there seems to be a Starbucks on every corner. 

### Acknowledgements
I also used sample code for my data preparation and analysis from Ms.Sconyers's Python intro document on jupyter notebooks. I recieved help from Ms.Sconyers also with the pandas cheat sheet. 

Cheat sheet: 
https://www.dataquest.io/blog/large_files/pandas-cheat-sheet.pdf

Again, I recieved my Starbucks Nutritional Data from the following source: https://www.kaggle.com/starbucks/starbucks-menu